### ISSUE TITLE
Implement brake in steering

### ISSUE THREAD

**Issue by irapha** [gtagency/buzzmobile#24]

**Comment by irapha** [comment]
This involves:
- adding `bool brake` to `CarPose.msg`, but I believe that was done already.
- Make `steering.py`'s tentacle calculation use a reasonable thickness, not just a single pixel width
  Please note that 'reasonable thickness' might mean calculating how many pixels BM is wide (a measurement we do not currently have) and change the mask creation for that number of pixels.
  (this might be a separate issue, but low-hanging fruit)

Also involves adding the following functionality to `steering.py`:
- create 'immediate future' mask, which is an arc at some distance from BM's position (using rosparams in `constants.yaml` such as `pixels_per_meter`)
- ANDs immediate future and chosen tentacle mask together
- uses result as mask to `lidar_model`
- sums result to have final score
- If score is greater than some threshold, set `CarPose`'s `break` to False, and send it with the calculated values of `velocity` and `angle`.
- Else send an empty `CarPose` msg with `break` set to True.

FInally, it involves having the `controller` (being worked on by @coletaylor788 in `actuators` branch), have similar functionality (if break button is on, send empty `CarPose` with break set to True, else, populate `CarPose`.)

**Comment by irapha** [comment]
The list above is inacurate. Someone ping me about updating it tomorrow please

**Comment by irapha** [comment]
Updated the comment

**Comment by coletaylor788** [comment]
@iRapha what do you mean by empty CarPose? Currently I am setting brake to true and velocity to 0. I am leaving the other inputs (steering angle and horn) as is. i.e. If you pressed brake and horn at the same time, it would brake and honk. Any concerns with that approach?

**Comment by irapha** [comment]
No concerns. That's exactly what I meant by empty

**Comment by irapha** [comment]
I started this on the `breaking` branch, sorry for the unfortunate branch name. Todos include:
- debugging steering because the tentacle being picked it too long. I believe this has to do with the `travel_distance` rosparam being too large.
- test braking functionality with new image_consts

**Comment by irapha** [comment]
Just updated travel_distance, and that's fixed now. I'm gonna create some test images (or run the lidar bag file we have) and see how it does, then create a PR

**Comment by irapha** [comment]
Testing blocked by #55 

As a note, another thing that's needed here is updating the architecture image.

**Comment by irapha** [comment]
Last break test to be run is using `some_lidar_data.bag`. This is blocked by #62.
Once that test is run, I'll make the PR.
